### PR TITLE
Introduce cluster 'team' setting and add to Member

### PR DESCRIPTION
**Pull request by raboof**
Introduced cluster-team.md so we can grow the documentation with each
PR, but did not add it to the ToC yet.

### CODE REVIEW

File: akka-cluster/src/main/resources/reference.conf [hunk 1]
```diff
@@ -65,6 +65,11 @@ akka {
     # move 'WeaklyUp' members to 'Up' status once convergence has been reached.
     allow-weakly-up-members = on
 
+    # Teams are used to make islands of the cluster that are colocated. It can be used
+    # to make the cluster "dc-aware", run the cluster in multiple availability zones or regions.
```
Comment (patriknw) [inline akka-cluster/src/main/resources/reference.conf]: spell it out data center aware, and run the cluster across multiple...

or do one say "run a cluster spanning multiple"?

File: akka-cluster/src/main/scala/akka/cluster/Member.scala [hunk 2]
```diff
@@ -22,6 +22,9 @@ class Member private[cluster] (
   val status:                    MemberStatus,
   val roles:                     Set[String]) extends Serializable {
 
+  lazy val team = roles.find(_.startsWith("team-"))
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/Member.scala]: explicit return type `: String`

File: akka-cluster/src/multi-jvm/scala/akka/cluster/MBeanSpec.scala [hunk 3]
```diff
@@ -120,28 +120,32 @@ abstract class MBeanSpec
...
+             |        "team-default"
              |      ],
              |      "status": "Up"
              |    },
              |    {
              |      "address": "${sortedNodes(3)}",
              |      "roles": [
-             |        "testNode"
+             |        "testNode",
+             |        "team-default"
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/MBeanSpec.scala]: are these sorted? should they be sorted to have deterministic result?

File: akka-docs/src/main/paradox/scala/cluster-team.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Cluster Team
+
+@@@ note
+
+Cluster teams are a work-in-progress feature, and behavior is still expected to change.
+
+@@@
+
+Teams are used to define islands of the cluster that are colocated.
+They can be used to make the cluster "dc-aware", run the cluster in multiple availability zones or regions.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/scala/cluster-team.md]: spell it out

File: akka-docs/src/main/paradox/scala/cluster-team.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+Cluster nodes can be assigned to a team by setting the `akka.cluster.team` setting.
+When no team is specified, a node will belong to the 'default' team.
+
+The team is added to the list of roles of the node with the prefix "team-".
```
Comment (patriknw) [inline akka-docs/src/main/paradox/scala/cluster-team.md]: good, let's add docs here, at least sketchy notes and then polish it at the end

**Review by patriknw — CHANGES_REQUESTED** [review]
looking good, just some nitpicks

**Review by johanandren — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Refs #23228

**Pull request merged**